assets/js/modules/sign-in-with-google/components/settings/SettingsView.js

```diff
@@ -0,0 +1,216 @@
+/**
+ * Sign in with Google Settings View component.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { createInterpolateElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useDispatch, useSelect } from 'googlesitekit-data';
+import {
+	MODULES_SIGN_IN_WITH_GOOGLE,
+	SIGN_IN_WITH_GOOGLE_SHAPES,
+	SIGN_IN_WITH_GOOGLE_TEXTS,
+	SIGN_IN_WITH_GOOGLE_THEMES,
+} from '../../datastore/constants';
+import StoreErrorNotices from '../../../../components/StoreErrorNotices';
+import DisplaySetting from '../../../../components/DisplaySetting';
+import Link from '../../../../components/Link';
+import SettingsNotice, {
+	TYPE_WARNING,
+} from '../../../../components/SettingsNotice';
+import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import WarningIcon from '../../../../../svg/icons/warning.svg';
+
+export default function SettingsView() {
+	const { dismissItem } = useDispatch( CORE_USER );
+
+	const clientID = useSelect( ( select ) =>
+		select( MODULES_SIGN_IN_WITH_GOOGLE ).getClientID()
+	);
+
+	const anyoneCanRegister = useSelect( ( select ) =>
+		select( CORE_SITE ).getAnyoneCanRegister()
+	);
+
+	const buttonShapeLabel = useSelect( ( select ) => {
+		const shape = select( MODULES_SIGN_IN_WITH_GOOGLE ).getShape();
+
+		return SIGN_IN_WITH_GOOGLE_SHAPES.find( ( { value } ) => {
+			return value === shape;
+		} )?.label;
+	} );
+
+	const buttonTextLabel = useSelect( ( select ) => {
+		const text = select( MODULES_SIGN_IN_WITH_GOOGLE ).getText();
+
+		return SIGN_IN_WITH_GOOGLE_TEXTS.find( ( { value } ) => {
+			return value === text;
+		} )?.label;
+	} );
+
+	const buttonThemeLabel = useSelect( ( select ) => {
+		const theme = select( MODULES_SIGN_IN_WITH_GOOGLE ).getTheme();
+
+		return SIGN_IN_WITH_GOOGLE_THEMES.find( ( { value } ) => {
+			return value === theme;
+		} )?.label;
+	} );
+
+	const oneTapEnabled = useSelect( ( select ) =>
+		select( MODULES_SIGN_IN_WITH_GOOGLE ).getOneTapEnabled()
+	);
+
+	const generalSettingsURL = useSelect(
+		( select ) =>
+			`${ select( CORE_SITE ).getAdminURL() }options-general.php`
+	);
```

Comment:
That's odd, I thought we could do this with this selector, but we can't without using some legacy behavior I guess. Let's at least construct this more properly :)

```suggestion
	const generalSettingsURL = useSelect(
		( select ) =>
			new URL( 'options-general.php', select( CORE_SITE ).getAdminURL() ).href
	);
```